express-delete/index.js

```diff
@@ -0,0 +1,46 @@
+const express = require('express');
+
+const grades = {
+  12: {
+    id: 12,
+    name: 'Tim Berners-Lee',
+    course: 'HTML',
+    score: 95
+  },
+  47: {
+    id: 47,
+    name: 'Brendan Eich',
+    course: 'JavaScript',
+    score: 100
+  },
+  273: {
+    id: 273,
+    name: 'Forbes Lindsay',
+    course: 'JavaScript',
+    score: 92
+  }
+};
+
+const app = express();
+
+app.get('/api/grades', (req, res) => {
+  const gradesArray = [];
+  for (const prop in grades) {
+    gradesArray.push(grades[prop]);
+  }
+  res.json(gradesArray);
+});
+
+app.delete('/api/grades/:id', (req, res) => {
+  for (const prop in grades) {
+    if (req.params.id === grades[prop].id.toString()) {
+      delete grades[prop];
+    }
+  }
```

Comment:
When you already know the property name you don't need a loop to find it. You can just do something like:

```JAVASCRIPT
const id = req.params.id;
delete grades[id];
```

If you want to check if the `id` is valid first you can do something like:

```JAVASCRIPT
const id = req.params.id;

if(grades[id]) {
  delete grades[id];
  return res.sendStatus(204);
} else {
  return res.sendStatus(404);
}
```